lib/block_view.go

```diff
@@ -6768,6 +6821,21 @@ func (bav *UtxoView) _connectCreateNFT(
 	if postEntry.IsNFT {
 		return 0, 0, nil, RuleErrorCreateNFTOnPostThatAlreadyIsNFT
 	}
+	if isBuyNow && blockHeight < BuyNowNFTBlockHeight {
```

Comment:
We should remove this check and instead blockHeight restrict the chunk of code that gets the IsBuyNowKey from extraData.